cd /data/NCATS_ifx/data/mRNASeq/ISB009 && mkdir -p bam/Sample_VJ4001-10_S24 && STAR --runThreadN $SLURM_CPUS_PER_TASK --genomeDir /fdb/STAR_current/GENCODE/Gencode_human/release_27/genes-75 --sjdbOverhang 75 --outSAMunmapped Within --outFilterType BySJout --outFilterMultimapNmax 20 --outFilterMismatchNmax 999 --outFilterMismatchNoverLmax 0.04 --alignIntronMin 20 --alignIntronMax 1000000 --alignMatesGapMax 1000000 --alignSJoverhangMin 8 --alignSJDBoverhangMin 1 --sjdbScore 1 --readFilesIn /data/NCATS_ifx/data/mRNASeq/ISB009/VJ4001-10_S24_1P.fastq.gz /data/NCATS_ifx/data/mRNASeq/ISB009/VJ4001-10_S24_2P.fastq.gz --readFilesCommand zcat --outSAMtype BAM SortedByCoordinate --outFileNamePrefix bam/Sample_VJ4001-10_S24/VJ4001-10_S24_hg38

cd /data/NCATS_ifx/data/mRNASeq/ISB009 && mkdir -p bam/Sample_VJ4001-11_S25 && STAR --runThreadN $SLURM_CPUS_PER_TASK --genomeDir /fdb/STAR_current/GENCODE/Gencode_human/release_27/genes-75 --sjdbOverhang 75 --outSAMunmapped Within --outFilterType BySJout --outFilterMultimapNmax 20 --outFilterMismatchNmax 999 --outFilterMismatchNoverLmax 0.04 --alignIntronMin 20 --alignIntronMax 1000000 --alignMatesGapMax 1000000 --alignSJoverhangMin 8 --alignSJDBoverhangMin 1 --sjdbScore 1 --readFilesIn /data/NCATS_ifx/data/mRNASeq/ISB009/VJ4001-11_S25_1P.fastq.gz /data/NCATS_ifx/data/mRNASeq/ISB009/VJ4001-11_S25_2P.fastq.gz --readFilesCommand zcat --outSAMtype BAM SortedByCoordinate --outFileNamePrefix bam/Sample_VJ4001-11_S25/VJ4001-11_S25_hg38

cd /data/NCATS_ifx/data/mRNASeq/ISB009 && mkdir -p bam/Sample_VJ4001-12_S26 && STAR --runThreadN $SLURM_CPUS_PER_TASK --genomeDir /fdb/STAR_current/GENCODE/Gencode_human/release_27/genes-75 --sjdbOverhang 75 --outSAMunmapped Within --outFilterType BySJout --outFilterMultimapNmax 20 --outFilterMismatchNmax 999 --outFilterMismatchNoverLmax 0.04 --alignIntronMin 20 --alignIntronMax 1000000 --alignMatesGapMax 1000000 --alignSJoverhangMin 8 --alignSJDBoverhangMin 1 --sjdbScore 1 --readFilesIn /data/NCATS_ifx/data/mRNASeq/ISB009/VJ4001-12_S26_1P.fastq.gz /data/NCATS_ifx/data/mRNASeq/ISB009/VJ4001-12_S26_2P.fastq.gz --readFilesCommand zcat --outSAMtype BAM SortedByCoordinate --outFileNamePrefix bam/Sample_VJ4001-12_S26/VJ4001-12_S26_hg38

cd /data/NCATS_ifx/data/mRNASeq/ISB009 && mkdir -p bam/Sample_VJ4001-13_S27 && STAR --runThreadN $SLURM_CPUS_PER_TASK --genomeDir /fdb/STAR_current/GENCODE/Gencode_human/release_27/genes-75 --sjdbOverhang 75 --outSAMunmapped Within --outFilterType BySJout --outFilterMultimapNmax 20 --outFilterMismatchNmax 999 --outFilterMismatchNoverLmax 0.04 --alignIntronMin 20 --alignIntronMax 1000000 --alignMatesGapMax 1000000 --alignSJoverhangMin 8 --alignSJDBoverhangMin 1 --sjdbScore 1 --readFilesIn /data/NCATS_ifx/data/mRNASeq/ISB009/VJ4001-13_S27_1P.fastq.gz /data/NCATS_ifx/data/mRNASeq/ISB009/VJ4001-13_S27_2P.fastq.gz --readFilesCommand zcat --outSAMtype BAM SortedByCoordinate --outFileNamePrefix bam/Sample_VJ4001-13_S27/VJ4001-13_S27_hg38

cd /data/NCATS_ifx/data/mRNASeq/ISB009 && mkdir -p bam/Sample_VJ4001-14_S28 && STAR --runThreadN $SLURM_CPUS_PER_TASK --genomeDir /fdb/STAR_current/GENCODE/Gencode_human/release_27/genes-75 --sjdbOverhang 75 --outSAMunmapped Within --outFilterType BySJout --outFilterMultimapNmax 20 --outFilterMismatchNmax 999 --outFilterMismatchNoverLmax 0.04 --alignIntronMin 20 --alignIntronMax 1000000 --alignMatesGapMax 1000000 --alignSJoverhangMin 8 --alignSJDBoverhangMin 1 --sjdbScore 1 --readFilesIn /data/NCATS_ifx/data/mRNASeq/ISB009/VJ4001-14_S28_1P.fastq.gz /data/NCATS_ifx/data/mRNASeq/ISB009/VJ4001-14_S28_2P.fastq.gz --readFilesCommand zcat --outSAMtype BAM SortedByCoordinate --outFileNamePrefix bam/Sample_VJ4001-14_S28/VJ4001-14_S28_hg38

cd /data/NCATS_ifx/data/mRNASeq/ISB009 && mkdir -p bam/Sample_VJ4001-15_S29 && STAR --runThreadN $SLURM_CPUS_PER_TASK --genomeDir /fdb/STAR_current/GENCODE/Gencode_human/release_27/genes-75 --sjdbOverhang 75 --outSAMunmapped Within --outFilterType BySJout --outFilterMultimapNmax 20 --outFilterMismatchNmax 999 --outFilterMismatchNoverLmax 0.04 --alignIntronMin 20 --alignIntronMax 1000000 --alignMatesGapMax 1000000 --alignSJoverhangMin 8 --alignSJDBoverhangMin 1 --sjdbScore 1 --readFilesIn /data/NCATS_ifx/data/mRNASeq/ISB009/VJ4001-15_S29_1P.fastq.gz /data/NCATS_ifx/data/mRNASeq/ISB009/VJ4001-15_S29_2P.fastq.gz --readFilesCommand zcat --outSAMtype BAM SortedByCoordinate --outFileNamePrefix bam/Sample_VJ4001-15_S29/VJ4001-15_S29_hg38

cd /data/NCATS_ifx/data/mRNASeq/ISB009 && mkdir -p bam/Sample_VJ4001-1_S15 && STAR --runThreadN $SLURM_CPUS_PER_TASK --genomeDir /fdb/STAR_current/GENCODE/Gencode_human/release_27/genes-75 --sjdbOverhang 75 --outSAMunmapped Within --outFilterType BySJout --outFilterMultimapNmax 20 --outFilterMismatchNmax 999 --outFilterMismatchNoverLmax 0.04 --alignIntronMin 20 --alignIntronMax 1000000 --alignMatesGapMax 1000000 --alignSJoverhangMin 8 --alignSJDBoverhangMin 1 --sjdbScore 1 --readFilesIn /data/NCATS_ifx/data/mRNASeq/ISB009/VJ4001-1_S15_1P.fastq.gz /data/NCATS_ifx/data/mRNASeq/ISB009/VJ4001-1_S15_2P.fastq.gz --readFilesCommand zcat --outSAMtype BAM SortedByCoordinate --outFileNamePrefix bam/Sample_VJ4001-1_S15/VJ4001-1_S15_hg38

cd /data/NCATS_ifx/data/mRNASeq/ISB009 && mkdir -p bam/Sample_VJ4001-2_S16 && STAR --runThreadN $SLURM_CPUS_PER_TASK --genomeDir /fdb/STAR_current/GENCODE/Gencode_human/release_27/genes-75 --sjdbOverhang 75 --outSAMunmapped Within --outFilterType BySJout --outFilterMultimapNmax 20 --outFilterMismatchNmax 999 --outFilterMismatchNoverLmax 0.04 --alignIntronMin 20 --alignIntronMax 1000000 --alignMatesGapMax 1000000 --alignSJoverhangMin 8 --alignSJDBoverhangMin 1 --sjdbScore 1 --readFilesIn /data/NCATS_ifx/data/mRNASeq/ISB009/VJ4001-2_S16_1P.fastq.gz /data/NCATS_ifx/data/mRNASeq/ISB009/VJ4001-2_S16_2P.fastq.gz --readFilesCommand zcat --outSAMtype BAM SortedByCoordinate --outFileNamePrefix bam/Sample_VJ4001-2_S16/VJ4001-2_S16_hg38

cd /data/NCATS_ifx/data/mRNASeq/ISB009 && mkdir -p bam/Sample_VJ4001-3_S17 && STAR --runThreadN $SLURM_CPUS_PER_TASK --genomeDir /fdb/STAR_current/GENCODE/Gencode_human/release_27/genes-75 --sjdbOverhang 75 --outSAMunmapped Within --outFilterType BySJout --outFilterMultimapNmax 20 --outFilterMismatchNmax 999 --outFilterMismatchNoverLmax 0.04 --alignIntronMin 20 --alignIntronMax 1000000 --alignMatesGapMax 1000000 --alignSJoverhangMin 8 --alignSJDBoverhangMin 1 --sjdbScore 1 --readFilesIn /data/NCATS_ifx/data/mRNASeq/ISB009/VJ4001-3_S17_1P.fastq.gz /data/NCATS_ifx/data/mRNASeq/ISB009/VJ4001-3_S17_2P.fastq.gz --readFilesCommand zcat --outSAMtype BAM SortedByCoordinate --outFileNamePrefix bam/Sample_VJ4001-3_S17/VJ4001-3_S17_hg38

cd /data/NCATS_ifx/data/mRNASeq/ISB009 && mkdir -p bam/Sample_VJ4001-4_S18 && STAR --runThreadN $SLURM_CPUS_PER_TASK --genomeDir /fdb/STAR_current/GENCODE/Gencode_human/release_27/genes-75 --sjdbOverhang 75 --outSAMunmapped Within --outFilterType BySJout --outFilterMultimapNmax 20 --outFilterMismatchNmax 999 --outFilterMismatchNoverLmax 0.04 --alignIntronMin 20 --alignIntronMax 1000000 --alignMatesGapMax 1000000 --alignSJoverhangMin 8 --alignSJDBoverhangMin 1 --sjdbScore 1 --readFilesIn /data/NCATS_ifx/data/mRNASeq/ISB009/VJ4001-4_S18_1P.fastq.gz /data/NCATS_ifx/data/mRNASeq/ISB009/VJ4001-4_S18_2P.fastq.gz --readFilesCommand zcat --outSAMtype BAM SortedByCoordinate --outFileNamePrefix bam/Sample_VJ4001-4_S18/VJ4001-4_S18_hg38

cd /data/NCATS_ifx/data/mRNASeq/ISB009 && mkdir -p bam/Sample_VJ4001-5_S19 && STAR --runThreadN $SLURM_CPUS_PER_TASK --genomeDir /fdb/STAR_current/GENCODE/Gencode_human/release_27/genes-75 --sjdbOverhang 75 --outSAMunmapped Within --outFilterType BySJout --outFilterMultimapNmax 20 --outFilterMismatchNmax 999 --outFilterMismatchNoverLmax 0.04 --alignIntronMin 20 --alignIntronMax 1000000 --alignMatesGapMax 1000000 --alignSJoverhangMin 8 --alignSJDBoverhangMin 1 --sjdbScore 1 --readFilesIn /data/NCATS_ifx/data/mRNASeq/ISB009/VJ4001-5_S19_1P.fastq.gz /data/NCATS_ifx/data/mRNASeq/ISB009/VJ4001-5_S19_2P.fastq.gz --readFilesCommand zcat --outSAMtype BAM SortedByCoordinate --outFileNamePrefix bam/Sample_VJ4001-5_S19/VJ4001-5_S19_hg38

cd /data/NCATS_ifx/data/mRNASeq/ISB009 && mkdir -p bam/Sample_VJ4001-6_S20 && STAR --runThreadN $SLURM_CPUS_PER_TASK --genomeDir /fdb/STAR_current/GENCODE/Gencode_human/release_27/genes-75 --sjdbOverhang 75 --outSAMunmapped Within --outFilterType BySJout --outFilterMultimapNmax 20 --outFilterMismatchNmax 999 --outFilterMismatchNoverLmax 0.04 --alignIntronMin 20 --alignIntronMax 1000000 --alignMatesGapMax 1000000 --alignSJoverhangMin 8 --alignSJDBoverhangMin 1 --sjdbScore 1 --readFilesIn /data/NCATS_ifx/data/mRNASeq/ISB009/VJ4001-6_S20_1P.fastq.gz /data/NCATS_ifx/data/mRNASeq/ISB009/VJ4001-6_S20_2P.fastq.gz --readFilesCommand zcat --outSAMtype BAM SortedByCoordinate --outFileNamePrefix bam/Sample_VJ4001-6_S20/VJ4001-6_S20_hg38

cd /data/NCATS_ifx/data/mRNASeq/ISB009 && mkdir -p bam/Sample_VJ4001-7_S21 && STAR --runThreadN $SLURM_CPUS_PER_TASK --genomeDir /fdb/STAR_current/GENCODE/Gencode_human/release_27/genes-75 --sjdbOverhang 75 --outSAMunmapped Within --outFilterType BySJout --outFilterMultimapNmax 20 --outFilterMismatchNmax 999 --outFilterMismatchNoverLmax 0.04 --alignIntronMin 20 --alignIntronMax 1000000 --alignMatesGapMax 1000000 --alignSJoverhangMin 8 --alignSJDBoverhangMin 1 --sjdbScore 1 --readFilesIn /data/NCATS_ifx/data/mRNASeq/ISB009/VJ4001-7_S21_1P.fastq.gz /data/NCATS_ifx/data/mRNASeq/ISB009/VJ4001-7_S21_2P.fastq.gz --readFilesCommand zcat --outSAMtype BAM SortedByCoordinate --outFileNamePrefix bam/Sample_VJ4001-7_S21/VJ4001-7_S21_hg38

cd /data/NCATS_ifx/data/mRNASeq/ISB009 && mkdir -p bam/Sample_VJ4001-8_S22 && STAR --runThreadN $SLURM_CPUS_PER_TASK --genomeDir /fdb/STAR_current/GENCODE/Gencode_human/release_27/genes-75 --sjdbOverhang 75 --outSAMunmapped Within --outFilterType BySJout --outFilterMultimapNmax 20 --outFilterMismatchNmax 999 --outFilterMismatchNoverLmax 0.04 --alignIntronMin 20 --alignIntronMax 1000000 --alignMatesGapMax 1000000 --alignSJoverhangMin 8 --alignSJDBoverhangMin 1 --sjdbScore 1 --readFilesIn /data/NCATS_ifx/data/mRNASeq/ISB009/VJ4001-8_S22_1P.fastq.gz /data/NCATS_ifx/data/mRNASeq/ISB009/VJ4001-8_S22_2P.fastq.gz --readFilesCommand zcat --outSAMtype BAM SortedByCoordinate --outFileNamePrefix bam/Sample_VJ4001-8_S22/VJ4001-8_S22_hg38

cd /data/NCATS_ifx/data/mRNASeq/ISB009 && mkdir -p bam/Sample_VJ4001-9_S23 && STAR --runThreadN $SLURM_CPUS_PER_TASK --genomeDir /fdb/STAR_current/GENCODE/Gencode_human/release_27/genes-75 --sjdbOverhang 75 --outSAMunmapped Within --outFilterType BySJout --outFilterMultimapNmax 20 --outFilterMismatchNmax 999 --outFilterMismatchNoverLmax 0.04 --alignIntronMin 20 --alignIntronMax 1000000 --alignMatesGapMax 1000000 --alignSJoverhangMin 8 --alignSJDBoverhangMin 1 --sjdbScore 1 --readFilesIn /data/NCATS_ifx/data/mRNASeq/ISB009/VJ4001-9_S23_1P.fastq.gz /data/NCATS_ifx/data/mRNASeq/ISB009/VJ4001-9_S23_2P.fastq.gz --readFilesCommand zcat --outSAMtype BAM SortedByCoordinate --outFileNamePrefix bam/Sample_VJ4001-9_S23/VJ4001-9_S23_hg38

